### PR TITLE
chore(frontend): Integrate run functions with KFP v2 API

### PR DESCRIPTION
**Pull request by jlyaoyuli**
The main purpose of this PR is integrate run functionalities with v2 API.

Primary proto difference between v1 and v2
1. from `id` to `run_id`
2. from `name` to `display_name`
3. `pipeline_spec` only store either `workflow_manifest` or `pipeline_manifest` in v2
4. `resource_reference` is deprecated in v2
5.  run details information is no longer wrapped in `run` field in the API response
6. `runtime_config` is move out from `pipeline_spec`

Note:
Some experiment-related API call will be integrated with v2 after recurring run integration.

### CODE REVIEW

File: frontend/src/components/viewers/MetricsDropdown.test.tsx [hunk 1]
```diff
@@ -16,16 +16,16 @@
...
 import { render, screen, waitFor, fireEvent } from '@testing-library/react';
 import * as React from 'react';
-import { CommonTestWrapper } from 'src/TestWrapper';
-import TestUtils, { testBestPractices } from 'src/TestUtils';
-import { Artifact, Event, Execution, Value } from 'src/third_party/mlmd';
-import * as metricsVisualizations from 'src/components/viewers/MetricsVisualizations';
-import * as Utils from 'src/lib/Utils';
-import { SelectedArtifact } from 'src/pages/CompareV2';
-import { LinkedArtifact } from 'src/mlmd/MlmdUtils';
+import { CommonTestWrapper } from '../../TestWrapper';
```
Comment (zijianjoy) [inline frontend/src/components/viewers/MetricsDropdown.test.tsx:19]: Please use absolute path instead of relative path in the import reference.

File: frontend/src/lib/Apis.ts [hunk 2]
```diff
@@ -17,7 +17,8 @@ import { ExperimentServiceApi, FetchAPI } from '../apis/experiment';
 import { ExperimentServiceApi as ExperimentServiceApiV2 } from '../apisv2beta1/experiment';
 import { JobServiceApi } from '../apis/job';
 import { ApiPipeline, ApiPipelineVersion, PipelineServiceApi } from '../apis/pipeline';
-import { RunServiceApi } from '../apis/run';
+import { RunServiceApi as RunServiceApiV1 } from '../apis/run';
```
Comment (zijianjoy) [inline frontend/src/lib/Apis.ts:20]: Please use absolute path instead of relative path in the import reference.

File: frontend/src/lib/StatusUtils.ts [hunk 3]
```diff
@@ -116,3 +129,76 @@ function wasNodeCached(node: NodeStatus): boolean {
...
+    case V2beta1RuntimeState.FAILED: // Fall through
+    case V2beta1RuntimeState.CANCELED:
+      return true;
+    case V2beta1RuntimeState.PENDING: // Fall through
+    case V2beta1RuntimeState.RUNNING: // Fall through
+    case V2beta1RuntimeState.CANCELING: // Fall through
+    case V2beta1RuntimeState.RUNTIMESTATEUNSPECIFIED:
+      return false;
+    default:
+      return false;
```
Comment (zijianjoy) [inline frontend/src/lib/StatusUtils.ts:147]: I think we should throw exception or have some sort of warnings instead, this is an unexpected situation.

File: frontend/src/lib/StatusUtils.ts [hunk 4]
```diff
@@ -41,6 +42,18 @@ export enum NodePhase {
...
+  [V2beta1RuntimeState.RUNTIMESTATEUNSPECIFIED, 'Unknown'],
+  [V2beta1RuntimeState.PENDING, 'Pending'],
+  [V2beta1RuntimeState.RUNNING, 'Running'],
+  [V2beta1RuntimeState.SUCCEEDED, 'Succeeded'],
+  [V2beta1RuntimeState.SKIPPED, 'Skipped'],
+  [V2beta1RuntimeState.FAILED, 'Failed'],
+  [V2beta1RuntimeState.CANCELING, 'Canceling'],
+  [V2beta1RuntimeState.CANCELED, 'Canceled'],
+  [V2beta1RuntimeState.PAUSED, 'Paused'],
+]);
```
Comment (zijianjoy) [inline frontend/src/lib/StatusUtils.ts:55]: What happen if more runtime states are introduced? How do we track this information and update accordingly?

File: frontend/src/pages/AllRunsList.test.tsx [hunk 5]
```diff
@@ -16,7 +16,7 @@
 
 import { shallow, ShallowWrapper } from 'enzyme';
 import * as React from 'react';
-import { ApiRunStorageState } from '../apis/run';
+import { V2beta1RunStorageState } from '../apisv2beta1/run';
```
Comment (zijianjoy) [inline frontend/src/pages/AllRunsList.test.tsx:19]: Please use absolute path instead of relative path in the import reference.

File: frontend/src/lib/Utils.tsx [hunk 6]
```diff
@@ -134,6 +142,12 @@ export function getRunDurationFromApiRun(apiRun?: ApiRun): string {
   return getDuration(new Date(apiRun.created_at), new Date(apiRun.finished_at));
 }
 
+export function getRunDurationFromRunV2(run?: V2beta1Run): string {
```
Comment (zijianjoy) [inline frontend/src/lib/Utils.tsx:145]: It looks duplicated with getRunDurationV2() above

File: frontend/src/lib/v2/CompareUtils.test.ts [hunk 7]
```diff
@@ -14,38 +14,34 @@
...
+import { testBestPractices } from '../../TestUtils';
 import {
   getScalarTableProps,
   getParamsTableProps,
   getValidRocCurveArtifactData,
   RunArtifact,
 } from './CompareUtils';
-import { Artifact, Event, Execution, Value } from 'src/third_party/mlmd';
-import { LinkedArtifact } from 'src/mlmd/MlmdUtils';
+import { Artifact, Event, Execution, Value } from '../../third_party/mlmd';
```
Comment (zijianjoy) [inline frontend/src/lib/v2/CompareUtils.test.ts:24]: Please use absolute path instead of relative path in the import reference.

File: frontend/src/pages/CompareV2.test.tsx [hunk 8]
```diff
@@ -16,18 +16,18 @@
...
 import { render, screen, waitFor, fireEvent } from '@testing-library/react';
 import * as React from 'react';
-import { CommonTestWrapper } from 'src/TestWrapper';
-import TestUtils, { testBestPractices } from 'src/TestUtils';
-import { Artifact, Context, Event, Execution } from 'src/third_party/mlmd';
-import { Apis } from 'src/lib/Apis';
-import { QUERY_PARAMS } from 'src/components/Router';
-import * as mlmdUtils from 'src/mlmd/MlmdUtils';
-import * as Utils from 'src/lib/Utils';
+import { CommonTestWrapper } from '../TestWrapper';
```
Comment (zijianjoy) [inline frontend/src/pages/CompareV2.test.tsx:19]: Please use absolute path instead of relative path in the import reference.

File: frontend/src/pages/RunList.test.tsx [hunk 9]
```diff
@@ -16,24 +16,16 @@
...
-import { ApiFilter, PredicateOp } from '../apis/filter';
-import {
-  ApiRun,
-  ApiRunDetail,
-  ApiResourceType,
-  ApiRunMetric,
-  RunMetricFormat,
-  ApiRunStorageState,
-} from '../apis/run';
+import { V2beta1Filter, V2beta1PredicateOperation } from '../apisv2beta1/filter';
```
Comment (zijianjoy) [inline frontend/src/pages/RunList.test.tsx:23]: Please use absolute path instead of relative path in the import reference.

File: frontend/src/pages/RunList.tsx [hunk 10]
```diff
@@ -47,14 +46,15 @@ interface RecurringRunInfo {
 interface DisplayRun {
   experiment?: ExperimentInfo;
   recurringRun?: RecurringRunInfo;
-  run: ApiRun;
+  run: V2beta1Run;
```
Comment (zijianjoy) [inline frontend/src/pages/RunList.tsx:49]: We still want to show V1 Runs. Please confirm if we can still support listing V1 Runs here.

File: frontend/src/pages/RunList.tsx [hunk 11]
```diff
@@ -47,14 +46,15 @@ interface RecurringRunInfo {
...
   pipelineVersion?: PipelineVersionInfo;
   error?: string;
 }
 
 interface DisplayMetric {
   metadata?: MetricMetadata;
-  metric?: ApiRunMetric;
+  // run metric field is currently not supported in v2 API
```
Comment (zijianjoy) [inline frontend/src/pages/RunList.tsx:56]: I think we need to hide it from official documentation as well. The change here is going to make metrics disappear on both V1 and V2. If we decide to hide metrics for both V1 and V2, can you go update https://www.kubeflow.org/docs/components/pipelines/v1/sdk/pipelines-metrics/#view-the-metrics to hide this feature as well?

File: frontend/src/pages/RunList.tsx [hunk 12]
```diff
@@ -488,29 +468,8 @@ class RunList extends React.PureComponent<RunListProps, RunListState> {
...
-          displayRun.error = 'Failed to get associated pipeline: ' + (await errorToMessage(err));
-          return;
-        }
-      }
-      displayRun.pipelineVersion = {
-        displayName: pipelineName,
-        pipelineId,
-        usePlaceholder: false,
-        versionId: undefined,
-      };
```
Comment (zijianjoy) [inline frontend/src/pages/RunList.tsx:509]: Why are this part deleted?

File: frontend/src/lib/StatusUtils.ts [hunk 13]
```diff
@@ -116,3 +129,76 @@ function wasNodeCached(node: NodeStatus): boolean {
...
+): V2beta1RuntimeState | undefined {
+  // Argo considers terminated runs as having "Failed", so we have to examine the failure message to
+  // determine why the run failed.
+  if (state === V2beta1RuntimeState.FAILED && nodeMessage === 'terminated') {
+    state = V2beta1RuntimeState.CANCELED;
+  }
+  return state;
+}
+
+export function parseNodePhaseV2(node: NodeStatus): NodePhase {
```
Comment (zijianjoy) [inline frontend/src/lib/StatusUtils.ts:188]: Where is this function being used? Are the logic here and below functions still applicable to V2 scenario?

File: frontend/src/pages/RunList.tsx (hunk 12 above)
Comment (jlyaoyuli) [reply frontend/src/pages/RunList.tsx:509]: Because `pipelineId` is no longer store in `pipeline_spec` or other field in `run` when using v2 API. That is, we are not able to get `pipelineId` from getRun or listRun response.

File: frontend/src/lib/Utils.tsx (hunk 6 above)
Comment (jlyaoyuli) [reply frontend/src/lib/Utils.tsx:145]: Great catch. Will remove this one.

File: frontend/src/lib/StatusUtils.ts (hunk 13 above)
Comment (jlyaoyuli) [reply frontend/src/lib/StatusUtils.ts:188]: Removed it.

File: frontend/src/lib/StatusUtils.ts (hunk 4 above)
Comment (jlyaoyuli) [reply frontend/src/lib/StatusUtils.ts:55]: The map is built according to v2 proto. All of the runtime states we currently have is included. If any status not existed in the proto is appeared, we will display `-` instead. That being said, any proto changes will lead to update for this map.

File: frontend/src/lib/StatusUtils.ts (hunk 3 above)
Comment (jlyaoyuli) [reply frontend/src/lib/StatusUtils.ts:147]: Good suggestion. Add a warn and an error to handle any unexpected state.

**Review by zijianjoy — COMMENTED** [review]
/lgtm

Pending for https://github.com/kubeflow/website/pull/3462 and https://github.com/kubeflow/website/pull/3463 to merge first.

**Comment by zijianjoy** [comment]
/lgtm
/approve

Thank you for the good work!
